### PR TITLE
Replace to try-with-resources

### PR DESCRIPTION
**Pull request by wreulicke**
Replace to try-with-resources

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: spotbugs/src/main/java/edu/umd/cs/findbugs/ba/SourceFile.java [hunk 1]
```diff
@@ -189,32 +189,25 @@ private synchronized void loadFileData() throws IOException {
...
-            mapBuilder.eof();
-
-            setData(out.toByteArray());
-        } finally {
-            if (in != null) {
-                in.close();
-            }
-        }
+	
+		try (InputStream in = dataSource.open()) {
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/ba/SourceFile.java]: Inconsistent indentation (tabs vs spaces)

File: spotbugs/src/main/java/edu/umd/cs/findbugs/config/UserPreferences.java [hunk 2]
```diff
@@ -175,21 +175,12 @@ public void read() {
...
-            try {
-                if (prefStream != null) {
-                    prefStream.close();
-                }
-            } catch (IOException ioe) {
-                // Ignore
-            }
         }
-
+        // Ignore
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/config/UserPreferences.java]: We don't need this line

File: spotbugs/src/main/java/edu/umd/cs/findbugs/ba/interproc/PropertyDatabase.java [hunk 3]
```diff
@@ -222,14 +211,8 @@ public void write(@WillClose OutputStream out) throws IOException {
...
-
-            try {
-                if (writer != null) {
-                    writer.close();
-                }
-            } catch (IOException e) {
-                // Ignore
-            }
+        
+            // Ignore
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/ba/interproc/PropertyDatabase.java]: Don't need this line.

File: spotbugs/src/main/java/edu/umd/cs/findbugs/config/UserPreferences.java (hunk 2 above)
Comment (wreulicke) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/config/UserPreferences.java]: Removed

File: spotbugs/src/main/java/edu/umd/cs/findbugs/ba/interproc/PropertyDatabase.java (hunk 3 above)
Comment (wreulicke) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/ba/interproc/PropertyDatabase.java]: Removed

File: spotbugs/src/main/java/edu/umd/cs/findbugs/ba/SourceFile.java (hunk 1 above)
Comment (wreulicke) [reply spotbugs/src/main/java/edu/umd/cs/findbugs/ba/SourceFile.java]: fixed

**Pull request merged**